ddtrace/internal/telemetry/writer.py

```diff
@@ -327,6 +329,14 @@ def disable(self):
         else:
             self.status = ServiceStatus.STOPPED
 
+    def enable_agentless_client(self, enabled=True):
+        # type: (bool) -> None
+
+        if self._client._agentless == enabled:
+            return
+
+        self._client = _TelemetryClient(enabled)
```

Comment:
What's the role of the agent when collection telemetry events? Should we make the default mode agent-less? Would we lose meaningful data?

Reply:
I think it make sense to start agentless if the API key is present during the tracer writer init time.
The only way we'd lose data is if the agent connection isn't initialized within 10s, but that's the same case as today and to my knowledge, it hasn't been a problem. If it is or ever becomes a problem, we can introduce some retry mechanism in the `periodic` function - something along the lines of adding a failure counter to each event and if we fail, we increment the counter and retry up to an agreed-upon threshold to prevent unbounded growth of event. 

For now I think I can just add my first suggestion if that sounds reasonable to you.

lmk

Reply:
Yes that sounds good. 

Also  in `TelemetryWriter._init__()` we can update the initialization of the agentless field to:  `agentless = config._dd_api_key is not None if agentless is None else agentless` (we should make this more readable 😓).   

With this approach we will always enable the agentless mode if `DD_API_KEY` is set. Otherwise we will default to sending telemetry to the agent. This should reduce the occurrence of dropped events. We can also do this in a follow up PR.